src/se/troed/plugin/Courier/Courier.java

```diff
@@ -285,8 +285,12 @@ private void stopUpdateThread() {
     private void deliverMail() {
         // find first online player with undelivered mail
         // spawn new thread to deliver the mail
-        Player[] players = getServer().getOnlinePlayers();
-        for (Player player : players) {
+        Player[] playerArray = new Player[100];
```

Comment:
Well the doc I thought said it would take the playerarray and use it if the players all fit, otherwise it would reallocated so I gave it 100, which is probably overkill for most, possibly 10 should do..  I will consider your request on taking it over and pm you on bukkit.  Thanks for updating the plugin.